packages/airnode-utilities/src/caching/index.ts

```diff
@@ -0,0 +1,252 @@
+ *
+ * @param startingKey should take the form of `blockedRequestId-` and the full key could be `blockedRequestId-0x0000000...`
+ */
+const getKeys = (startingKey?: string): string[] => {
+  try {
+    const keys = readdirSync(CACHE_BASE_PATH);
+    if (startingKey) {
+      return keys.filter((key) => key.indexOf(startingKey) === 0);
+    }
+    return keys;
+  } catch (e) {
+    logger.error(`Unable to retrieve keys from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+
+  return [];
+};
+
+/**
+ * Get the value stored for a key (aka the contents of the file with the key as name).
+ *
+ * @param key the key whose contents should be returned
+ */
+const getValueForKey = (key: string): any | undefined => {
+  try {
+    if (!existsSync(join(CACHE_BASE_PATH, key))) {
+      return undefined;
+    }
+
+    return JSON.parse(readFileSync(join(CACHE_BASE_PATH, key)).toString());
+  } catch (e) {
+    logger.error(`Unable to read key data from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+
+  return undefined;
+};
+
+/**
+ * Remove a key, this is discouraged and should only be used by sweep operations (garbage collection).
+ *
+ * @param key the key to remove
+ */
+const removeKey = (key: string) => {
+  try {
+    // Force will not throw an error if the file doesn't exist
+    rmSync(join(CACHE_BASE_PATH, key), { force: true });
+  } catch (e) {
+    logger.error(`Unable to remove key from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Add a cache entry
+ *
+ * @param key the name of the key, ideally in the form of `something-{uniqueId}`, eg. `requestIdBlocked-0x0000...`
+ * @param data the data to store for the key, arbitrary
+ * @param force force overwriting of an existing key (discouraged)
+ */
+const addKey = (key: string, data: any, force = false) => {
+  // To not break existing tests do nothing if we're executing in a test environment.
+  if (isJest() && !force) {
+    return;
+  }
+
+  const filePath = join(CACHE_BASE_PATH, key);
+  try {
+    const fileExists = existsSync(filePath);
+    if (fileExists && !force) {
+      logger.error(`Unable to overwrite key from fs cache as key exists and force is set to false`);
+      logger.error(`Key: ${key}`);
+      return;
+    }
+
+    if (fileExists) {
+      rmSync(filePath, { force: true });
+    }
+
+    writeFileSync(filePath, JSON.stringify(data));
+  } catch (e) {
+    logger.error(`Unable to remove key from fs cache`);
+    logger.error((e as Error).stack!);
+  }
+};
+
+/**
+ * Get all keys and their associated values
+ * This is a potentially dangerous operation as it will cause the entire cache to be loaded into memory... which could
+ * be more than 500 MB.
+ *
+ * On Lambda RAM and tmpfs are separate resources, but on GCP tmpfs *is* RAM, so reading 500 MB of cache into RAM will
+ * incur a >1GB memory usage cumulatively.
+ */
+const getAllKeysAndValues = (): Record<string, any> => {
```

Comment:
Do we need to provide this? Especially since it has those disadvantages?

Reply:
It's removal doesn't impact me and the app should first do a key search, so you're right, I'll remove it :)